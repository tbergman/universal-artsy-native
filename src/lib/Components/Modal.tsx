import { theme } from "lib/Components/Bidding/Elements/Theme"
import React from "react"
import { Modal as RNModal, TouchableWithoutFeedback, View, ViewProperties } from "react-native"
import styled from "styled-components/native"
import { Sans14, SansMedium14 } from "./Bidding/Elements/Typography"
import { SecondaryOutlineButton } from "./Buttons"

interface ModalProps extends ViewProperties {
  headerText: string
  detailText: string
  visible?: boolean
  closeModal?: () => void
}

const ModalBackgroundView = styled.View`
  background-color: #00000099;
  flex: 1;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

const ModalInnerView = styled.View`
  width: 300;
  background-color: white;
  padding: 20px;
  opacity: 1;
  border-radius: 2px;
`

export class Modal extends React.Component<ModalProps, any> {
  constructor(props) {
    super(props)

    this.state = { modalVisible: props.visible || false }
  }

  componentDidUpdate(prevProps) {
    if (this.props.visible !== prevProps.visible) {
      this.setState({ modalVisible: this.props.visible })
    }
  }

  closeModal() {
    if (this.props.closeModal) {
      this.props.closeModal()
    }
    this.setState({ modalVisible: false })
  }

  render() {
    const { headerText, detailText } = this.props

    return (
      <View style={{ marginTop: 22 }}>
        <RNModal animationType="fade" transparent={true} visible={this.state.modalVisible}>
          <TouchableWithoutFeedback onPress={() => this.closeModal()}>
            <ModalBackgroundView>
              <TouchableWithoutFeedback onPress={null}>
                <ModalInnerView>
                  <View style={{ paddingBottom: 10 }}>
                    <SansMedium14>{headerText}</SansMedium14>
                  </View>
                  <View style={{ paddingBottom: 30 }}>
                    <Sans14 color={theme.colors.black60}>{detailText}</Sans14>
                  </View>
                  <SecondaryOutlineButton
                    text="Ok"
                    style={{ height: 40, borderRadius: 2, borderWidth: 2 }}
                    onPress={() => {
                      this.closeModal()
                    }}
                  />
                </ModalInnerView>
              </TouchableWithoutFeedback>
            </ModalBackgroundView>
          </TouchableWithoutFeedback>
        </RNModal>
      </View>
    )
  }
}
