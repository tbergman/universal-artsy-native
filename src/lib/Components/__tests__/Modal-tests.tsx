import React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import { Modal } from "../Modal"

it("looks like expected", () => {
  const tree = renderer
    .create(
      // tslint:disable-next-line:no-empty
      <Modal headerText="An error occurred" detailText="This is an error moop." />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
