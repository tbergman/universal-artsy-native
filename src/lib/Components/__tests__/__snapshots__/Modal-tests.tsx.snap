// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks like expected 1`] = `
<View
  style={
    Object {
      "marginTop": 22,
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#00000099",
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      testID={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 2,
              "opacity": 1,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
              "width": 300,
            },
            undefined,
          ]
        }
        testID={undefined}
      >
        <View
          style={
            Object {
              "paddingBottom": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            fontSize={2}
            lineHeight={5}
            style={
              Array [
                Object {
                  "fontFamily": "Unica77LL-Medium",
                  "fontSize": 16,
                  "lineHeight": 5,
                },
                undefined,
              ]
            }
          >
            An error occurred
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingBottom": 30,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            color="#666"
            ellipsizeMode="tail"
            fontSize={2}
            lineHeight={5}
            style={
              Array [
                Object {
                  "color": "#666",
                  "fontFamily": "Unica77LL-Regular",
                  "fontSize": 16,
                  "lineHeight": 5,
                },
                undefined,
              ]
            }
          >
            This is an error moop.
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 255, 255, 1)",
              "borderColor": "rgba(229, 229, 229, 1)",
              "borderRadius": 2,
              "borderWidth": 2,
              "height": 40,
              "justifyContent": "center",
              "opacity": 1,
            }
          }
          testID={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 12,
                },
                Object {
                  "fontFamily": "AvantGardeGothicITC",
                },
                Object {
                  "color": "rgba(0, 0, 0, 1)",
                  "fontFamily": "Unica77LL-Medium",
                  "fontSize": 14,
                  "opacity": 1,
                },
              ]
            }
          >
            Ok
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
